features/poller/shell/src/main/java/org/opennms/netmgt/poller/shell/Poll.java

```diff
@@ -195,6 +202,16 @@ public Object execute() throws Exception {
                         } else {
                             System.out.printf("(No properties were returned by the monitor.\n");
                         }
+                        if (pollStatus.getDeviceConfig() != null) {
+                            DeviceConfig deviceConfig = pollStatus.getDeviceConfig();
+                            System.out.printf("Received file %s with content .. \n\n", deviceConfig.getFilename());
+                            if (deviceConfig.getFilename().contains(".gz")) {
+                                // Decompress if this is compressed file
+                                byte[] dcBytes = DeviceConfigUtil.decompressGzipToBytes(deviceConfig.getContent());
```

Comment:
Same here. We need this every time we read/display the content. This should be part of the `DeviceConfig` or even done on the receiver.